src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -552,7 +552,9 @@ async def coursier_fetch_one_coord(
         targets = await Get(
             Targets,
             UnparsedAddressInputs(
-                [request.pants_address], owning_address=None, description_of_origin="TODO(#14468)"
+                [request.pants_address],
+                owning_address=None,
+                description_of_origin="<infallible - coursier fetch>",
```

Comment:
@tdyas @chrisjrn @stuhood I could use a double-check of this. My reasoning is that it seems Pants only adds the `pants_address` when it deems it is safe to do? It's generated by Pants, not by untrusted user input.

But is the `pants_address` saved in the lockfile saved to disk? Meaning if target definitions change, the lock might become invalid?

(Ditto that I can add a comment if helpful)

Reply:
The JVM lockfile usually only stores Maven coordinates, not target addresses. For example: 

```
# --- BEGIN PANTS LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---
# {
#   "version": 1,
#   "generated_with_requirements": [
#     "com.lihaoyi:acyclic_2.13:0.2.1,url=not_provided,jar=not_provided",
#     "org.scala-lang:scala-library:2.13.8,url=not_provided,jar=not_provided"
#   ]
# }
# --- END PANTS LOCKFILE METADATA ---
```

The `pants_address` apparently is added if the user provided a custom local jar; i.e., not downloaded via Coursier.